advanced-usage/index.md

```diff
@@ -326,26 +351,78 @@ addresses, and refresh the page to establish the desired connection.
+
+{{ easy_image(src="joystick-button-mapping", width=500, center=true) }}
+
+##### MAVLink `MANUAL_CONTROL` Messages
+
+[`MANUAL_CONTROL`](https://mavlink.io/en/messages/common.html#MANUAL_CONTROL) MAVLink messages are 
+automatically sent to the vehicle at 25Hz, which is not currently configurable.
+
+Button functions are determined by the autopilot firmware - e.g. in ArduSub they correspond to
+[`BTNn_FUNCTION`](https://docs.bluerobotics.com/ardusub-zola/software/autopilot/ArduSub-4.1/developers/parameters/#btnn-function-function-for-button)
+parameter values.
+
+As a few caveats:
+- ArduSub <= 4.1.x only supports 16 independent buttons and 4 axis ranges in its `MANUAL_CONTROL` handling
+    - More recent versions support the extended protocol, with 32 buttons and 6 motion axis inputs
+- ArduRover does not support buttons via the `MANUAL_CONTROL` protocol
+- When mapping the Z (vertical) motion axis range, note that ArduSub uses 0 to +1000 for full reverse to full
+  forwards, whereas other vehicle types use -1000 to +1000
+
+The function mapping decision process is designed to minimise intrusiveness to existing setups. When mapping a
+`MANUAL_CONTROL` button function to a joystick button, Cockpit
+1. First tries to use already mapped functions
+    - e.g. if there is already a `BTNn_FUNCTION` configured to enable Stabilize mode, a Cockpit button being set
+      to enable Stabilize mode will map to that existing function bit
+1. Then overwrites available Disabled `BTNn_FUNCTION`s
+1. Then tries to overwrite any mapped function bits that Cockpit currently isn't using
+    - If you try to change a Cockpit function to a `MANUAL_CONTROL` function when there are no buttons left it
+      will display an error message about insufficient buttons
+
+It is permitted to map the joystick button functions without a vehicle connected, in which case any relevant
+autopilot parameters will be automatically remapped once the vehicle connects. This may result in errors if
+more `MANUAL_CONTROL` button functions have been assigned than are supported by the vehicle.
+
+##### Cockpit Actions
+
+Joystick buttons can also be configured to run more general functionalities, like modifying the interface or
+sending a single MAVLink message. The current supported Actions are:
+
+- `go_to_next_view`
+- `go_to_previous_view`
+- `toggle_bottom_bar`
+- `toggle_full_screen`
+- `mavlink_arm`
+- `mavlink_disarm`
+
+##### Modifier Keys
+
+Modifiers allow sacrificing one button in order to add an extra functionality slot for every non-modifier button.
+Pressing a button while a modifier is held runs the modified function instead of the regular button function.
+
+Currently only a single modifier is available (Shift), which is also mapped to the MAVLink `MANUAL_CONTROL` shift
+functionality (activating `BTNn_SFUNCTION`s when the shifted function is configured to a MAVLink `MANUAL_CONTROL`
+output, instead of the usual `BTNn_FUNCTION`s).
```

Comment:
"You can also use this same shift layout for other protocol functions as well (e.g.: Cockpit Actions). Mix them anyway you like. The only protocol which is contrained to only regular and shift is the MANUAL CONTROL."